images/base/files/usr/local/bin/entrypoint

```diff
@@ -104,10 +104,17 @@ fix_cgroup() {
   fi
   echo 'INFO: detected cgroup v1'
   echo 'INFO: fix cgroup mounts for all subsystems'
-  # see: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
-  # capture initial state before modifying
+  # See: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
+  # Capture initial state before modifying
+  #
+  # Basically we're looking for the cgroup-path for the cpu controller for the
+  # current process. this tells us what cgroup-path the container is in.
+  # Then we collect the subsystems that are active on this path.
+  # We assume the cpu controller is in use on all node containers.
+  #
+  # See: https://man7.org/linux/man-pages/man7/cgroups.7.html
   local current_cgroup
-  current_cgroup=$(grep systemd /proc/self/cgroup | cut -d: -f3)
+  current_cgroup=$(grep -E '^[^:]*:([^:]*,)?cpu(,[^,:]*)?:.*' /proc/self/cgroup | cut -d: -f3)
```

Comment:
arguably the first and last `*` should be `+` since these fields can't be empty AIUI, but there also shouldn't be more than 2 `:` anyhow and we match both non-optionally.

Reply:
sample of this:
```
root@175aed878411:/# cat /proc/self/cgroup
11:devices:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
10:pids:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
9:cpuset:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
8:rdma:/
7:freezer:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
6:memory:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
5:net_cls,net_prio:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
4:perf_event:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
3:blkio:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
2:cpu,cpuacct:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
1:name=systemd:/docker/175aed878411df93ce0cbd75904376da6e18318197872e27f1563491f41ffb58
```

Before we were naively matching on the `name=systemd` named heirarchy, which is a special alternative to a controller (it has no controllers, just for keeping track of stuff). This particular name is systemd specific.